File: ios/IssueTracker/IssueTracker/IssueList/Model/Label.swift
//
//  Label.swift
//  IssueTracker
//
//  Created by 지북 on 2021/06/09.
//

import Foundation

struct Label: Codable {
    var id: Int
    var name: String
    var description: String?
    var color: String
}

struct Labels: Codable {
    var labels: [Label]
}

File: be/issue-tracker-be/src/main/java/com/codesquad/issuetracker/label/domain/Label.java
package com.codesquad.issuetracker.label.domain;

import lombok.*;

@AllArgsConstructor
@Getter
@EqualsAndHashCode(of = "id")
@ToString
@Builder
public class Label {
    private Long id;
    private String name;
    private String description;
    private String color;

    public Label update(Label labelForUpdate) {
        name = labelForUpdate.getName();
        description = labelForUpdate.getDescription();
        color = labelForUpdate.getColor();

        return this;
    }
}
